django/contrib/postgres/lookups.py

```diff
@@ -74,3 +74,15 @@ def process_rhs(self, compiler, connection):
         result = super().process_rhs(compiler, connection)
         # Treat None lookup values as null.
         return ("'null'", []) if result == ('%s', [None]) else result
+
+
+class JSONIn(In):
+    def process_rhs(self, compiler, connection):
+        rhs, rhs_params = super().process_rhs(compiler, connection)
+        if not self.rhs_is_direct_value():
+            # We're doing an explicit type cast here as
+            # Postgres 9.3 doesn't have access to a
```

Comment:
The master branch only supports PostgreSQL 9.4+.

Reply:
Is it worth making the code conditional on a `connection.features` flag so that PostgreSQL 9.5+ can use  `to_jsonb`?

Reply:
According to [someone on the Internet](https://stackoverflow.com/questions/31969901/postgres-json-to-jsonb) there is a non-trivial performance cost when converting from `json` to `jsonb`, so we should avoid doing it unless it's necessary. Happy to make the change if you think the performance cost is going to make a difference here.

Reply:
Have added a check for Postgres 9.4. It's not explicitly tested atm but I assume in this case it's more of a job for the build system. Let me know if that looks acceptable.